addons/account/models/account_move_line.py

```diff
@@ -1359,6 +1359,8 @@ def init(self):
         """
         create_index(self._cr, 'account_move_line_partner_id_ref_idx', 'account_move_line', ["partner_id", "ref"])
         create_index(self._cr, 'account_move_line_date_name_id_idx', 'account_move_line', ["date desc", "move_name desc", "id"])
+        create_index(self._cr, 'account_move_line__unreconciled_index', 'account_move_line', ['account_id', 'partner_id'],
+                     where="(reconciled IS NULL OR reconciled = false OR reconciled IS NOT true) AND parent_state = 'posted'")
```

Comment:
This `where` looks overkill to me... `reconciled IS NOT TRUE AND parent_state = 'posted'` is enough I think :thinking:

Reply:
Actually, it is only to be safer with already existing queries elsewhere in the code, and with how the ORM converts the domains.
Otherwise the query planner can't use the index as it doesn't exactly match.